core/domain/opportunity_services.py

```diff
@@ -0,0 +1,380 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Commands that can be used to operate on opportunity models."""
+
+from constants import constants
+from core.domain import exp_fetchers
+from core.domain import opportunity_domain
+from core.domain import topic_services
+from core.platform import models
+import feconf
+
+(opportunity_models,) = models.Registry.import_models(
+    [models.NAMES.opportunity])
+
+
+def is_exploration_available_for_contribution(exp_id):
+    """Checks whether a given exploration id belongs to a curated list of
+    exploration i.e, whether it's used as the chapter of any story.
+
+    Note: Currently, the curated exploration are those lessons which are used in
+    the chapters of a story of a topic.
+
+    Args:
+        exp_id: str. The id of the exploration which is needed to be checked.
+
+    Returns:
+        bool. Whether the given exp_id belongs to the curated explorations.
+    """
+    model = opportunity_models.ExplorationOpportunitySummaryModel.get(
+        exp_id, strict=False)
+
+    return True if model is not None else False
```

Comment:
When an exploration is removed from a story, will its corresponding opportunity models be removed?

If the exploration is subsequently inserted back into the story, what happens to its opportunity models -- will they be regenerated from scratch? If so, is past contributor progress lost?